src/main/java/net/glowstone/dispenser/ArmorDispenseBehavior.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package net.glowstone.dispenser;
+
+import net.glowstone.block.GlowBlock;
+import net.glowstone.block.blocktype.BlockDispenser;
+import org.bukkit.Material;
+import org.bukkit.block.BlockFace;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.entity.Player;
+import java.util.List;
+
+/**
+ *
+ * @author cag45
+ */
+public class ArmorDispenseBehavior extends DefaultDispenseBehavior {
+    DefaultDispenseBehavior defaultBehavior = new DefaultDispenseBehavior();
+    
+    @Override
+    protected ItemStack dispenseStack(GlowBlock block, ItemStack stack) {
+        Material armor = stack.clone().getType();
+        BlockFace facing = BlockDispenser.getFacing(block);
+        GlowBlock target = block.getRelative(facing);
+        Location location1 = target.getLocation();
+        Location location2 = location1.clone();
+        location2.setY(location2.getY() - 1);
+        World world = location1.getWorld();
+        
+        switch (armor) { //Get the type of armor in stack
+            case LEATHER_HELMET: armor = Material.DIAMOND_HELMET; break;
+            case GOLD_HELMET: armor = Material.DIAMOND_HELMET; break;
+            case IRON_HELMET: armor = Material.DIAMOND_HELMET; break;
+            case CHAINMAIL_HELMET: armor = Material.DIAMOND_HELMET; break;
+            case LEATHER_BOOTS: armor = Material.DIAMOND_BOOTS; break;
+            case GOLD_BOOTS: armor = Material.DIAMOND_BOOTS; break;
+            case IRON_BOOTS: armor = Material.DIAMOND_BOOTS; break;
+            case CHAINMAIL_BOOTS: armor = Material.DIAMOND_BOOTS; break;
+            case LEATHER_LEGGINGS: armor = Material.DIAMOND_LEGGINGS; break;
+            case GOLD_LEGGINGS: armor = Material.DIAMOND_LEGGINGS; break;
+            case IRON_LEGGINGS: armor = Material.DIAMOND_LEGGINGS; break;
+            case CHAINMAIL_LEGGINGS: armor = Material.DIAMOND_LEGGINGS; break;
+            case LEATHER_CHESTPLATE: armor = Material.DIAMOND_CHESTPLATE; break;
+            case GOLD_CHESTPLATE: armor = Material.DIAMOND_CHESTPLATE; break;
+            case IRON_CHESTPLATE: armor = Material.DIAMOND_CHESTPLATE; break;
+            case CHAINMAIL_CHESTPLATE: armor = Material.DIAMOND_CHESTPLATE; break;
+        }
+        
+        List<Player> players = world.getPlayers();
```

Comment:
I think this functionality applies to other entities as well (those who can equip armor). You can use [`world.getNearbyEntities(location, x, y, z)`](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/World.html#getNearbyEntities(org.bukkit.Location,%20double,%20double,%20double)) to look through entities in an area.